vignettes/manual/Unitvals.Rmd

```diff
@@ -53,3 +53,18 @@ if( varName == D_PREINDUSTRIAL_N2O ) {`
             `H_ASSERT( data.date == Core::undefinedIndex() , "date not allowed" );`
             `N0 = unitval::parse_unitval( data.value_str, data.units_str, U_PPBV_N2O );`
 ```
+
+Missing values
+--------------
+Briefly, when you need to return a missing value, use the `MISSING_FLOAT` macro defined in `inst/include/unitval.hpp`.
+This macro will return `NA_REAL` (from `Rcpp`) if Hector is compiled as an R package (technically, if the `USE_RCPP` macro is defined) and `NAN` (from `std`) otherwise.
+
+Why this complexity?
+R (and, by extension, Rcpp) distinguish between two kinds of "placeholder" numeric values:
+A _missing_ value (`NA_real_` in R; often abbreviated to `NA`) represents an unobserved value; for instance, if you take annual observations from 1990 to 2000 but failed to make a measurement in 1996.
+On the other hand, _not a number_ (`NaN`) represents the result of an invalid mathematical operation, such as taking the square root of a negative number.
+Both of these values propagate through any calculations in which they are involved, both in R and C++, and both can be detected and removed by R's `is.na()` function and removed by `na.omit`, `na.rm = TRUE`, and friends.
```

Comment:
Note also that `is.na()` returns true for both `NA` and `NaN` values, while `is.nan()` returns true for `NaN` values, but false for `NA`.

Reply:
Added a note to this effect in 919fa30.